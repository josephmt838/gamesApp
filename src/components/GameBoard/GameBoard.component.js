import React from "react";
import PropTypes from "prop-types";

import "./GameBoard.scss";

const GameBoard = ({ children, classes }) => {
  return (
    <section
      className={`game-board ${classes ? classes : ""}`}
      data-testid={"classes"}
    >
      {children}
    </section>
  );
};

GameBoard.propTypes = {
  childre: PropTypes.node,
  classes: PropTypes.string
};

export default GameBoard;
