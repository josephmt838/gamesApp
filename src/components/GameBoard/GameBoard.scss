.game-board {
  width: 90%;
  height: 700px;
  margin: 20px auto;
  border: 1px solid darkblue;
  border-radius: 5px;
  overflow: scroll;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}
