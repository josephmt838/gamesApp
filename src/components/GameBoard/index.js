import GameBoard from './GameBoard.component';

export default GameBoard;