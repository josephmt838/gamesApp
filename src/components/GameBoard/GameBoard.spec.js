import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import GameBoard from "../GameBoard";

describe("GameBoard Component", () => {
  it("Renders GameBoard component", () => {
    const tree = renderer.create(<GameBoard classes="testing-jest" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Supplies classes to gameboard", () => {
    const outputClass = "game-board testing-jest";
    const newClass = "testing-jest";
    const game = <GameBoard classes={newClass} />;

    const { getByTestId } = render(game);

    const renderedClasses = Object.values(
      getByTestId("classes").classList
    ).join(" ");

    expect(renderedClasses).toEqual(outputClass);
  });

  it("Supplies children to GameBoard", () => {
    const game = (
      <GameBoard>
        <p data-testid="children">Testing Children</p>
      </GameBoard>
    );

    const { getByTestId } = render(game);
    expect(getByTestId("children").innerHTML).toBe("Testing Children");
  });
});
