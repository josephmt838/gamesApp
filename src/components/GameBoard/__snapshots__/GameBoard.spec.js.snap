// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoard Component Renders GameBoard component 1`] = `
<section
  className="game-board testing-jest"
  data-testid="classes"
/>
`;
